src/index.ts

```diff
@@ -43,18 +59,32 @@ function getFileName(path: string, file: StrapiFile) {
 }
 
 function makeBlobServiceClient(config: Config) {
-    const account = trimParam(config.account);
-    const accountKey = trimParam(config.accountKey);
-    const sasToken = trimParam(config.sasToken);
     const serviceBaseURL = getServiceBaseUrl(config);
-    // if accountKey doesn't contain value return below line
-    if (sasToken != '') {
-        const anonymousCredential = new AnonymousCredential();
-        return new BlobServiceClient(`${serviceBaseURL}${sasToken}`, anonymousCredential);
+
+    switch (config.auth_type) {
+        case 'default': {
+            const account = trimParam(config.account);
+            const accountKey = trimParam(config.accountKey);
+            const sasToken = trimParam(config.sasToken);
+            if (sasToken != '') {
+                const anonymousCredential = new AnonymousCredential();
+                return new BlobServiceClient(`${serviceBaseURL}${sasToken}`, anonymousCredential);
+            }
+            const sharedKeyCredential = new StorageSharedKeyCredential(account, accountKey);
+            const pipeline = newPipeline(sharedKeyCredential);
+            return new BlobServiceClient(serviceBaseURL, pipeline);
+        }
+        case 'msi': {
+            const clientId = trimParam(config.clientId);
+            if (clientId != null && clientId != '') {
+                return new BlobServiceClient(
+                    serviceBaseURL,
+                    new DefaultAzureCredential({ managedIdentityClientId: clientId })
+                );
+            }
+            return new BlobServiceClient(serviceBaseURL, new DefaultAzureCredential());
+        }
```

Comment:
We should just use the switch/cases default vs defining a case for it

```suggestion

    switch (config.auth_type) {
        case 'msi': {
            const clientId = trimParam(config.clientId);
            if (clientId != null && clientId != '') {
                return new BlobServiceClient(
                    serviceBaseURL,
                    new DefaultAzureCredential({ managedIdentityClientId: clientId })
                );
            }
            return new BlobServiceClient(serviceBaseURL, new DefaultAzureCredential());
        }
       default: {
            const account = trimParam(config.account);
            const accountKey = trimParam(config.accountKey);
            const sasToken = trimParam(config.sasToken);
            if (sasToken != '') {
                const anonymousCredential = new AnonymousCredential();
                return new BlobServiceClient(`${serviceBaseURL}${sasToken}`, anonymousCredential);
            }
            const sharedKeyCredential = new StorageSharedKeyCredential(account, accountKey);
            const pipeline = newPipeline(sharedKeyCredential);
            return new BlobServiceClient(serviceBaseURL, pipeline);
        }
}
```

Reply:
The explicit cases are what are doing the type narrowing on the tags, so moving them to the default case means TS doesn't recognise the params as belonging to that specific branch of the union. 

I can change the name from `default` to `accountKeyOrSas` if you find it more informative?

Reply:
Have added a never case to the default so there isn't a fall-through case. 

Not sure where the type checking happens on the strapi side, if at all?

Reply:
That makes sense. Thanks for adding the default case.